Support for `cohere` 5.0.0+

### Body

Cohere v5.0.0 just released and seems like it might not work out of the box with [Cohere Provider ](https://airflow.apache.org/docs/apache-airflow-providers-cohere/stable/index.html)

```console
airflow/providers/cohere/hooks/cohere.py:58: error: Unexpected keyword argument
"max_retries" for "Client"  [call-arg]
            return cohere.Client(
                   ^
/usr/local/lib/python3.8/site-packages/cohere/client.py:60: note: "Client" defined here
airflow/providers/cohere/hooks/cohere.py:58: error: Unexpected keyword argument
"api_url" for "Client"  [call-arg]
            return cohere.Client(
                   ^
/usr/local/lib/python3.8/site-packages/cohere/client.py:60: note: "Client" defined here
airflow/providers/cohere/hooks/cohere.py:56: error: Signature of "get_conn"
incompatible with supertype "BaseHook"  [override]
        def get_conn(self) -> cohere.Client:
        ^
airflow/providers/cohere/hooks/cohere.py:56: note:      Superclass:
airflow/providers/cohere/hooks/cohere.py:56: note:          def get_conn(self) -> Any
airflow/providers/cohere/hooks/cohere.py:56: note:      Subclass:
airflow/providers/cohere/hooks/cohere.py:56: note:          Client
airflow/providers/cohere/hooks/cohere.py:67: error: Incompatible return value
type (got "Union[List[List[float]], EmbedByTypeResponseEmbeddings]", expected
"List[List[float]]")  [return-value]
            return embeddings
                   ^~~~~~~~~~
airflow/providers/cohere/hooks/cohere.py:80: error: Too many positional
arguments for "generate" of "BaseCohere"  [misc]
                self.get_conn.generate("Test", max_tokens=10)
                ^~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~
```

We need eventually add support for v5, [Migration Guide](https://github.com/cohere-ai/cohere-python/blob/main/4.0.0-5.0.0-migration-guide.md)

### Committer

- [X] I acknowledge that I am a maintainer/committer of the Apache Airflow project.

--- a/generated/provider_dependencies.json
+++ b/generated/provider_dependencies.json
@@ -383,7 +383,7 @@
   "cohere": {
     "deps": [
       "apache-airflow>=2.9.0",
-      "cohere>=4.37,<5"
+      "cohere>=5.13.4"
     ],
     "devel-deps": [],
     "plugins": [],

--- a/providers/src/airflow/providers/cohere/hooks/cohere.py
+++ b/providers/src/airflow/providers/cohere/hooks/cohere.py
@@ -20,7 +20,7 @@ from __future__ import annotations
 import logging
 import warnings
 from functools import cached_property
-from typing import TYPE_CHECKING, Any
+from typing import TYPE_CHECKING, Any, cast
 
 import cohere
 
@@ -29,7 +29,7 @@ from airflow.hooks.base import BaseHook
 
 if TYPE_CHECKING:
     from cohere.core.request_options import RequestOptions
-    from cohere.types import ChatMessages, EmbedByTypeResponseEmbeddings
+    from cohere.types import ChatMessages, EmbedByTypeResponseEmbeddings, UserChatMessageV2
 
 
 logger = logging.getLogger(__name__)
@@ -111,9 +111,13 @@ class CohereHook(BaseHook):
     def test_connection(
         self,
         model: str = "command-r-plus-08-2024",
+        messages: list[UserChatMessageV2] | None = None,
     ) -> tuple[bool, str]:
         try:
-            messages: ChatMessages = [{"role": "user", "content": "hello world!"}]
+            if messages is None:
+                messages = [{"role": "user", "content": "hello world!"}]
+            else:
+                messages = cast(ChatMessages, messages)
             self.get_conn.chat(model=model, messages=messages)
             return True, "Connection successfully established."
         except Exception as e:

--- a/providers/src/airflow/providers/cohere/hooks/cohere.py
+++ b/providers/src/airflow/providers/cohere/hooks/cohere.py
@@ -16,10 +16,11 @@
 # under the License.
 
 from __future__ import annotations
-import warnings
+
 import logging
+import warnings
 from functools import cached_property
-from typing import Any, List
+from typing import Any
 
 import cohere
 
@@ -27,6 +28,7 @@ from airflow.exceptions import AirflowProviderDeprecationWarning
 from airflow.hooks.base import BaseHook
 
 logger = logging.getLogger(__name__)
+
 
 class CohereHook(BaseHook):
     """
@@ -57,33 +59,31 @@ class CohereHook(BaseHook):
         self.timeout = timeout
         self.max_retries = max_retries
         self.request_options = request_options
-        
+
         if self.max_retries is not None:
             warnings.warn(
                 "The 'max_retries' parameter is deprecated. Use 'request_options' with {'max_retries': value} instead.",
                 AirflowProviderDeprecationWarning,
-                stacklevel=2
+                stacklevel=2,
             )
             if self.request_options is None:
-                self.request_options = {'max_retries': self.max_retries}
+                self.request_options = {"max_retries": self.max_retries}
             else:
-                self.request_options['max_retries'] = self.max_retries
+                self.request_options["max_retries"] = self.max_retries
 
     @cached_property
     def get_conn(self) -> cohere.ClientV2:
-        """
-        :return: Cohere V2 client for API interaction.
-        """
+        ":return: Cohere V2 client for API interaction."
         conn = self.get_connection(self.conn_id)
         return cohere.ClientV2(
             api_key=conn.password,
             timeout=self.timeout,
             base_url=conn.host or None,
-            request_options=self.request_options
+            request_options=self.request_options,
         )
 
     def create_embeddings(
-        self, texts: List[str], model: str = "embed-multilingual-v3.0"
+        self, texts: list[str], model: str = "embed-multilingual-v3.0"
     ) -> list[list[float]]:
         """
         Create embeddings for the given texts using the specified model.
@@ -92,16 +92,14 @@ class CohereHook(BaseHook):
         :param model: The model to use for creating embeddings. Default is 'embed-multilingual-v3.0'.
         :return: List of embedding vectors, where each inner list represents an embedding.
         """
-        logger.info(f"Creating embeddings with model: {model}")
+        logger.info("Creating embeddings with model: embed-multilingual-v3.0")
         response = self.get_conn.embed(texts=texts, model=model, request_options=self.request_options)
         embeddings = response.embeddings
         return embeddings
 
     @classmethod
     def get_ui_field_behaviour(cls) -> dict[str, Any]:
-        """
-        :return: Dictionary defining the UI behavior for connection configuration in Airflow.
-        """
+        ":return: Dictionary defining the UI behavior for connection configuration in Airflow."
         return {
             "hidden_fields": ["schema", "login", "port", "extra"],
             "relabeling": {

--- a/providers/src/airflow/providers/cohere/hooks/cohere.py
+++ b/providers/src/airflow/providers/cohere/hooks/cohere.py
@@ -29,7 +29,7 @@ from airflow.hooks.base import BaseHook
 
 if TYPE_CHECKING:
     from cohere.core.request_options import RequestOptions
-    from cohere.types import EmbedByTypeResponseEmbeddings, UserChatMessageV2
+    from cohere.types import ChatMessages, EmbedByTypeResponseEmbeddings
 
 
 logger = logging.getLogger(__name__)
@@ -67,11 +67,14 @@ class CohereHook(BaseHook):
 
         if self.max_retries:
             warnings.warn(
-                "The 'max_retries' parameter is deprecated. Use 'request_options' with {'max_retries': value} instead.",
+                "Argument `max_retries` is deprecated. Use `request_options` dict for function-specific request configuration.",
                 AirflowProviderDeprecationWarning,
                 stacklevel=2,
             )
-            self.request_options['max_retries'] = self.max_retries
+            if self.request_options is None:
+                self.request_options = {"max_retries": self.max_retries}
+            else:
+                self.request_options.update({"max_retries": self.max_retries})
 
     @cached_property
     def get_conn(self) -> cohere.ClientV2:  # type: ignore[override]
@@ -80,7 +83,6 @@ class CohereHook(BaseHook):
             api_key=conn.password,
             timeout=self.timeout,
             base_url=conn.host or None,
-            request_options=self.request_options,
         )
 
     def create_embeddings(
@@ -106,8 +108,12 @@ class CohereHook(BaseHook):
             },
         }
 
-    def test_connection(self, model: str = "command-r-plus-08-2024", messages: UserChatMessageV2 = [{"role": "user", "content": "hello world!"}]) -> tuple[bool, str]:
+    def test_connection(
+        self,
+        model: str = "command-r-plus-08-2024",
+    ) -> tuple[bool, str]:
         try:
+            messages: ChatMessages = [{"role": "user", "content": "hello world!"}]
             self.get_conn.chat(model=model, messages=messages)
             return True, "Connection successfully established."
         except Exception as e:

--- a/providers/src/airflow/providers/cohere/hooks/cohere.py
+++ b/providers/src/airflow/providers/cohere/hooks/cohere.py
@@ -111,13 +111,11 @@ class CohereHook(BaseHook):
     def test_connection(
         self,
         model: str = "command-r-plus-08-2024",
-        messages: list[UserChatMessageV2] | None = None,
+        messages: ChatMessages | None = None,
     ) -> tuple[bool, str]:
         try:
             if messages is None:
-                messages = [{"role": "user", "content": "hello world!"}]
-            else:
-                messages = cast(ChatMessages, messages)
+                messages = [UserChatMessageV2(role="user", content="hello world!")]
             self.get_conn.chat(model=model, messages=messages)
             return True, "Connection successfully established."
         except Exception as e:

--- a/providers/src/airflow/providers/cohere/hooks/cohere.py
+++ b/providers/src/airflow/providers/cohere/hooks/cohere.py
@@ -1,4 +1,3 @@
-#
 # Licensed to the Apache Software Foundation (ASF) under one
 # or more contributor license agreements.  See the NOTICE file
 # distributed with this work for additional information
@@ -15,25 +14,30 @@
 # KIND, either express or implied.  See the License for the
 # specific language governing permissions and limitations
 # under the License.
-from __future__ import annotations
 
+from __future__ import annotations
+import warnings
+import logging
 from functools import cached_property
-from typing import Any
+from typing import Any, List
 
 import cohere
 
+from airflow.exceptions import AirflowProviderDeprecationWarning
 from airflow.hooks.base import BaseHook
 
+logger = logging.getLogger(__name__)
 
 class CohereHook(BaseHook):
     """
-    Use Cohere Python SDK to interact with Cohere platform.
+    Use Cohere Python SDK to interact with Cohere platform using API v2.
 
     .. seealso:: https://docs.cohere.com/docs
 
     :param conn_id: :ref:`Cohere connection id <howto/connection:cohere>`
-    :param timeout: Request timeout in seconds.
-    :param max_retries: Maximal number of retries for requests.
+    :param timeout: Request timeout in seconds. Optional.
+    :param max_retries: Maximal number of retries for requests. Deprecated, use request_options instead. Optional.
+    :param request_options: Dictionary for function-specific request configuration. Optional.
     """
 
     conn_name_attr = "conn_id"
@@ -46,28 +50,58 @@ class CohereHook(BaseHook):
         conn_id: str = default_conn_name,
         timeout: int | None = None,
         max_retries: int | None = None,
+        request_options: dict | None = None,
     ) -> None:
         super().__init__()
         self.conn_id = conn_id
         self.timeout = timeout
         self.max_retries = max_retries
+        self.request_options = request_options
+        
+        if self.max_retries is not None:
+            warnings.warn(
+                "The 'max_retries' parameter is deprecated. Use 'request_options' with {'max_retries': value} instead.",
+                AirflowProviderDeprecationWarning,
+                stacklevel=2
+            )
+            if self.request_options is None:
+                self.request_options = {'max_retries': self.max_retries}
+            else:
+                self.request_options['max_retries'] = self.max_retries
 
     @cached_property
-    def get_conn(self) -> cohere.Client:  # type: ignore[override]
+    def get_conn(self) -> cohere.ClientV2:
+        """
+        :return: Cohere V2 client for API interaction.
+        """
         conn = self.get_connection(self.conn_id)
-        return cohere.Client(
-            api_key=conn.password, timeout=self.timeout, max_retries=self.max_retries, api_url=conn.host
+        return cohere.ClientV2(
+            api_key=conn.password,
+            timeout=self.timeout,
+            base_url=conn.host or None,
+            request_options=self.request_options
         )
 
     def create_embeddings(
-        self, texts: list[str], model: str = "embed-multilingual-v2.0"
+        self, texts: List[str], model: str = "embed-multilingual-v3.0"
     ) -> list[list[float]]:
-        response = self.get_conn.embed(texts=texts, model=model)
+        """
+        Create embeddings for the given texts using the specified model.
+
+        :param texts: List of texts to create embeddings for.
+        :param model: The model to use for creating embeddings. Default is 'embed-multilingual-v3.0'.
+        :return: List of embedding vectors, where each inner list represents an embedding.
+        """
+        logger.info(f"Creating embeddings with model: {model}")
+        response = self.get_conn.embed(texts=texts, model=model, request_options=self.request_options)
         embeddings = response.embeddings
         return embeddings
 
     @classmethod
     def get_ui_field_behaviour(cls) -> dict[str, Any]:
+        """
+        :return: Dictionary defining the UI behavior for connection configuration in Airflow.
+        """
         return {
             "hidden_fields": ["schema", "login", "port", "extra"],
             "relabeling": {
@@ -76,8 +110,13 @@ class CohereHook(BaseHook):
         }
 
     def test_connection(self) -> tuple[bool, str]:
+        """
+        Test the connection to Cohere by attempting a chat request.
+
+        :return: Tuple containing a boolean indicating success and a message.
+        """
         try:
-            self.get_conn.generate("Test", max_tokens=10)
-            return True, "Connection established"
+            self.get_conn.chat("Test")
+            return True, "Connection successfully established."
         except Exception as e:
-            return False, str(e)
+            return False, f"Unexpected error: {str(e)}"

--- a/providers/src/airflow/providers/cohere/hooks/cohere.py
+++ b/providers/src/airflow/providers/cohere/hooks/cohere.py
@@ -20,16 +20,17 @@ from __future__ import annotations
 import logging
 import warnings
 from functools import cached_property
-from typing import TYPE_CHECKING, Any, cast
+from typing import TYPE_CHECKING, Any
 
 import cohere
+from cohere.types import UserChatMessageV2
 
 from airflow.exceptions import AirflowProviderDeprecationWarning
 from airflow.hooks.base import BaseHook
 
 if TYPE_CHECKING:
     from cohere.core.request_options import RequestOptions
-    from cohere.types import ChatMessages, EmbedByTypeResponseEmbeddings, UserChatMessageV2
+    from cohere.types import ChatMessages, EmbedByTypeResponseEmbeddings
 
 
 logger = logging.getLogger(__name__)

--- a/providers/src/airflow/providers/cohere/hooks/cohere.py
+++ b/providers/src/airflow/providers/cohere/hooks/cohere.py
@@ -20,12 +20,17 @@ from __future__ import annotations
 import logging
 import warnings
 from functools import cached_property
-from typing import Any
+from typing import TYPE_CHECKING, Any
 
 import cohere
 
 from airflow.exceptions import AirflowProviderDeprecationWarning
 from airflow.hooks.base import BaseHook
+
+if TYPE_CHECKING:
+    from cohere.core.request_options import RequestOptions
+    from cohere.types import EmbedByTypeResponseEmbeddings
+
 
 logger = logging.getLogger(__name__)
 
@@ -52,7 +57,7 @@ class CohereHook(BaseHook):
         conn_id: str = default_conn_name,
         timeout: int | None = None,
         max_retries: int | None = None,
-        request_options: dict | None = None,
+        request_options: RequestOptions | None = None,
     ) -> None:
         super().__init__()
         self.conn_id = conn_id
@@ -72,8 +77,7 @@ class CohereHook(BaseHook):
                 self.request_options["max_retries"] = self.max_retries
 
     @cached_property
-    def get_conn(self) -> cohere.ClientV2:
-        ":return: Cohere V2 client for API interaction."
+    def get_conn(self) -> cohere.ClientV2:  # type: ignore[override]
         conn = self.get_connection(self.conn_id)
         return cohere.ClientV2(
             api_key=conn.password,
@@ -84,22 +88,20 @@ class CohereHook(BaseHook):
 
     def create_embeddings(
         self, texts: list[str], model: str = "embed-multilingual-v3.0"
-    ) -> list[list[float]]:
-        """
-        Create embeddings for the given texts using the specified model.
-
-        :param texts: List of texts to create embeddings for.
-        :param model: The model to use for creating embeddings. Default is 'embed-multilingual-v3.0'.
-        :return: List of embedding vectors, where each inner list represents an embedding.
-        """
+    ) -> EmbedByTypeResponseEmbeddings:
         logger.info("Creating embeddings with model: embed-multilingual-v3.0")
-        response = self.get_conn.embed(texts=texts, model=model, request_options=self.request_options)
+        response = self.get_conn.embed(
+            texts=texts,
+            model=model,
+            input_type="search_document",
+            embedding_types=["float"],
+            request_options=self.request_options,
+        )
         embeddings = response.embeddings
         return embeddings
 
     @classmethod
     def get_ui_field_behaviour(cls) -> dict[str, Any]:
-        ":return: Dictionary defining the UI behavior for connection configuration in Airflow."
         return {
             "hidden_fields": ["schema", "login", "port", "extra"],
             "relabeling": {
@@ -107,14 +109,9 @@ class CohereHook(BaseHook):
             },
         }
 
-    def test_connection(self) -> tuple[bool, str]:
-        """
-        Test the connection to Cohere by attempting a chat request.
-
-        :return: Tuple containing a boolean indicating success and a message.
-        """
+    def test_connection(self, model: str = "command-r-plus-08-2024") -> tuple[bool, str]:
         try:
-            self.get_conn.chat("Test")
+            self.get_conn.chat(model=model, messages=[{"role": "user", "content": "hello world!"}])
             return True, "Connection successfully established."
         except Exception as e:
             return False, f"Unexpected error: {str(e)}"

--- a/providers/src/airflow/providers/cohere/hooks/cohere.py
+++ b/providers/src/airflow/providers/cohere/hooks/cohere.py
@@ -29,7 +29,7 @@ from airflow.hooks.base import BaseHook
 
 if TYPE_CHECKING:
     from cohere.core.request_options import RequestOptions
-    from cohere.types import EmbedByTypeResponseEmbeddings
+    from cohere.types import EmbedByTypeResponseEmbeddings, UserChatMessageV2
 
 
 logger = logging.getLogger(__name__)
@@ -65,16 +65,13 @@ class CohereHook(BaseHook):
         self.max_retries = max_retries
         self.request_options = request_options
 
-        if self.max_retries is not None:
+        if self.max_retries:
             warnings.warn(
                 "The 'max_retries' parameter is deprecated. Use 'request_options' with {'max_retries': value} instead.",
                 AirflowProviderDeprecationWarning,
                 stacklevel=2,
             )
-            if self.request_options is None:
-                self.request_options = {"max_retries": self.max_retries}
-            else:
-                self.request_options["max_retries"] = self.max_retries
+            self.request_options['max_retries'] = self.max_retries
 
     @cached_property
     def get_conn(self) -> cohere.ClientV2:  # type: ignore[override]
@@ -109,9 +106,9 @@ class CohereHook(BaseHook):
             },
         }
 
-    def test_connection(self, model: str = "command-r-plus-08-2024") -> tuple[bool, str]:
+    def test_connection(self, model: str = "command-r-plus-08-2024", messages: UserChatMessageV2 = [{"role": "user", "content": "hello world!"}]) -> tuple[bool, str]:
         try:
-            self.get_conn.chat(model=model, messages=[{"role": "user", "content": "hello world!"}])
+            self.get_conn.chat(model=model, messages=messages)
             return True, "Connection successfully established."
         except Exception as e:
             return False, f"Unexpected error: {str(e)}"

--- a/providers/src/airflow/providers/cohere/operators/embedding.py
+++ b/providers/src/airflow/providers/cohere/operators/embedding.py
@@ -26,6 +26,7 @@ from airflow.providers.cohere.hooks.cohere import CohereHook
 
 if TYPE_CHECKING:
     from cohere.core.request_options import RequestOptions
+    from cohere.types import EmbedByTypeResponseEmbeddings
 
     from airflow.utils.context import Context
 
@@ -86,6 +87,6 @@ class CohereEmbeddingOperator(BaseOperator):
             request_options=self.request_options,
         )
 
-    def execute(self, context: Context) -> list[list[float]]:
+    def execute(self, context: Context) -> EmbedByTypeResponseEmbeddings:
         """Embed texts using Cohere embed services."""
         return self.hook.create_embeddings(self.input_text)

--- a/providers/src/airflow/providers/cohere/operators/embedding.py
+++ b/providers/src/airflow/providers/cohere/operators/embedding.py
@@ -17,20 +17,20 @@
 
 from __future__ import annotations
 
-from collections.abc import Sequence
 from functools import cached_property
-from typing import TYPE_CHECKING, Any
+from typing import TYPE_CHECKING, Any, Sequence
 
 from airflow.models import BaseOperator
 from airflow.providers.cohere.hooks.cohere import CohereHook
 
 if TYPE_CHECKING:
+    from cohere.core.request_options import RequestOptions
+
     from airflow.utils.context import Context
 
 
 class CohereEmbeddingOperator(BaseOperator):
-    """
-    Creates the embedding base by interacting with cohere hosted services.
+    """Creates the embedding base by interacting with cohere hosted services.
 
     .. seealso::
         For more information on how to use this operator, take a look at the guide:
@@ -41,6 +41,17 @@ class CohereEmbeddingOperator(BaseOperator):
         information for Cohere. Defaults to "cohere_default".
     :param timeout: Timeout in seconds for Cohere API.
     :param max_retries: Number of times to retry before failing.
+    :param request_options: Request-specific configuration.
+        Fields:
+        - timeout_in_seconds: int. The number of seconds to await an API call before timing out.
+
+        - max_retries: int. The max number of retries to attempt if the API call fails.
+
+        - additional_headers: typing.Dict[str, typing.Any]. A dictionary containing additional parameters to spread into the request's header dict
+
+        - additional_query_parameters: typing.Dict[str, typing.Any]. A dictionary containing additional parameters to spread into the request's query parameters dict
+
+        - additional_body_parameters: typing.Dict[str, typing.Any]. A dictionary containing additional parameters to spread into the request's body parameters dict
     """
 
     template_fields: Sequence[str] = ("input_text",)
@@ -51,6 +62,7 @@ class CohereEmbeddingOperator(BaseOperator):
         conn_id: str = CohereHook.default_conn_name,
         timeout: int | None = None,
         max_retries: int | None = None,
+        request_options: RequestOptions | None = None,
         **kwargs: Any,
     ):
         super().__init__(**kwargs)
@@ -60,11 +72,17 @@ class CohereEmbeddingOperator(BaseOperator):
         self.input_text = input_text
         self.timeout = timeout
         self.max_retries = max_retries
+        self.request_options = request_options
 
     @cached_property
     def hook(self) -> CohereHook:
         """Return an instance of the CohereHook."""
-        return CohereHook(conn_id=self.conn_id, timeout=self.timeout, max_retries=self.max_retries)
+        return CohereHook(
+            conn_id=self.conn_id,
+            timeout=self.timeout,
+            max_retries=self.max_retries,
+            request_options=self.request_options,
+        )
 
     def execute(self, context: Context) -> list[list[float]]:
         """Embed texts using Cohere embed services."""

--- a/providers/src/airflow/providers/cohere/operators/embedding.py
+++ b/providers/src/airflow/providers/cohere/operators/embedding.py
@@ -17,8 +17,9 @@
 
 from __future__ import annotations
 
+from collections.abc import Sequence
 from functools import cached_property
-from typing import TYPE_CHECKING, Any, Sequence
+from typing import TYPE_CHECKING, Any
 
 from airflow.models import BaseOperator
 from airflow.providers.cohere.hooks.cohere import CohereHook
@@ -30,7 +31,8 @@ if TYPE_CHECKING:
 
 
 class CohereEmbeddingOperator(BaseOperator):
-    """Creates the embedding base by interacting with cohere hosted services.
+    """
+    Creates the embedding base by interacting with cohere hosted services.
 
     .. seealso::
         For more information on how to use this operator, take a look at the guide:

--- a/providers/src/airflow/providers/cohere/provider.yaml
+++ b/providers/src/airflow/providers/cohere/provider.yaml
@@ -47,7 +47,7 @@ integrations:
 
 dependencies:
   - apache-airflow>=2.9.0
-  - cohere>=4.37,<5
+  - cohere>=5.13.4
 
 hooks:
   - integration-name: Cohere

--- a/providers/tests/cohere/hooks/test_cohere.py
+++ b/providers/tests/cohere/hooks/test_cohere.py
@@ -31,19 +31,19 @@ class TestCohereHook:
 
     def test__get_api_key(self):
         api_key = "test"
-        api_url = "http://some_host.com"
+        base_url = "http://some_host.com"
         timeout = 150
         max_retries = 5
         with (
             patch.object(
                 CohereHook,
                 "get_connection",
-                return_value=Connection(conn_type="cohere", password=api_key, host=api_url),
+                return_value=Connection(conn_type="cohere", password=api_key, host=base_url),
             ),
-            patch("cohere.Client") as client,
+            patch("cohere.ClientV2") as client,
         ):
             hook = CohereHook(timeout=timeout, max_retries=max_retries)
             _ = hook.get_conn
             client.assert_called_once_with(
-                api_key=api_key, timeout=timeout, max_retries=max_retries, api_url=api_url
+                api_key=api_key, timeout=timeout, max_retries=max_retries, api_url=base_url
             )

--- a/providers/tests/cohere/hooks/test_cohere.py
+++ b/providers/tests/cohere/hooks/test_cohere.py
@@ -33,15 +33,14 @@ class TestCohereHook:
         api_key = "test"
         base_url = "http://some_host.com"
         timeout = 150
-        request_options = None
         with (
             patch.object(
                 CohereHook,
                 "get_connection",
-                return_value=Connection(conn_type="cohere", password=api_key, host=base_url, request_options=request_options),
+                return_value=Connection(conn_type="cohere", password=api_key, host=base_url),
             ),
             patch("cohere.ClientV2") as client,
         ):
             hook = CohereHook(timeout=timeout)
             _ = hook.get_conn
-            client.assert_called_once_with(api_key=api_key, timeout=timeout, base_url=base_url, request_options=request_options)
+            client.assert_called_once_with(api_key=api_key, timeout=timeout, base_url=base_url)

--- a/providers/tests/cohere/hooks/test_cohere.py
+++ b/providers/tests/cohere/hooks/test_cohere.py
@@ -33,7 +33,6 @@ class TestCohereHook:
         api_key = "test"
         base_url = "http://some_host.com"
         timeout = 150
-        max_retries = 5
         with (
             patch.object(
                 CohereHook,
@@ -42,8 +41,6 @@ class TestCohereHook:
             ),
             patch("cohere.ClientV2") as client,
         ):
-            hook = CohereHook(timeout=timeout, max_retries=max_retries)
+            hook = CohereHook(timeout=timeout)
             _ = hook.get_conn
-            client.assert_called_once_with(
-                api_key=api_key, timeout=timeout, max_retries=max_retries, api_url=base_url
-            )
+            client.assert_called_once_with(api_key=api_key, timeout=timeout, base_url=base_url)

--- a/providers/tests/cohere/hooks/test_cohere.py
+++ b/providers/tests/cohere/hooks/test_cohere.py
@@ -33,14 +33,15 @@ class TestCohereHook:
         api_key = "test"
         base_url = "http://some_host.com"
         timeout = 150
+        request_options = None
         with (
             patch.object(
                 CohereHook,
                 "get_connection",
-                return_value=Connection(conn_type="cohere", password=api_key, host=base_url),
+                return_value=Connection(conn_type="cohere", password=api_key, host=base_url, request_options=request_options),
             ),
             patch("cohere.ClientV2") as client,
         ):
             hook = CohereHook(timeout=timeout)
             _ = hook.get_conn
-            client.assert_called_once_with(api_key=api_key, timeout=timeout, base_url=base_url)
+            client.assert_called_once_with(api_key=api_key, timeout=timeout, base_url=base_url, request_options=request_options)

--- a/providers/tests/cohere/operators/test_embedding.py
+++ b/providers/tests/cohere/operators/test_embedding.py
@@ -38,14 +38,15 @@ def test_cohere_embedding_operator(cohere_client, get_connection):
     base_url = "http://some_host.com"
     timeout = 150
     texts = ["On Kernel-Target Alignment. We describe a family of global optimization procedures"]
+    request_options = None
 
     get_connection.return_value = Connection(conn_type="cohere", password=api_key, host=base_url)
     client_obj = MagicMock()
     cohere_client.return_value = client_obj
     client_obj.embed.return_value = resp
 
-    op = CohereEmbeddingOperator(task_id="embed", conn_id="some_conn", input_text=texts, timeout=timeout)
+    op = CohereEmbeddingOperator(task_id="embed", conn_id="some_conn", input_text=texts, timeout=timeout, request_options=request_options)
 
     val = op.execute(context={})
-    cohere_client.assert_called_once_with(api_key=api_key, base_url=base_url, timeout=timeout)
+    cohere_client.assert_called_once_with(api_key=api_key, base_url=base_url, timeout=timeout, request_options=request_options)
     assert val == embedded_obj

--- a/providers/tests/cohere/operators/test_embedding.py
+++ b/providers/tests/cohere/operators/test_embedding.py
@@ -45,8 +45,14 @@ def test_cohere_embedding_operator(cohere_client, get_connection):
     cohere_client.return_value = client_obj
     client_obj.embed.return_value = resp
 
-    op = CohereEmbeddingOperator(task_id="embed", conn_id="some_conn", input_text=texts, timeout=timeout, request_options=request_options)
+    op = CohereEmbeddingOperator(
+        task_id="embed",
+        conn_id="some_conn",
+        input_text=texts,
+        timeout=timeout,
+        request_options=request_options,
+    )
 
     val = op.execute(context={})
-    cohere_client.assert_called_once_with(api_key=api_key, base_url=base_url, timeout=timeout, request_options=request_options)
+    cohere_client.assert_called_once_with(api_key=api_key, base_url=base_url, timeout=timeout)
     assert val == embedded_obj

--- a/providers/tests/cohere/operators/test_embedding.py
+++ b/providers/tests/cohere/operators/test_embedding.py
@@ -23,7 +23,7 @@ from airflow.providers.cohere.operators.embedding import CohereEmbeddingOperator
 
 
 @patch("airflow.providers.cohere.hooks.cohere.CohereHook.get_connection")
-@patch("cohere.Client")
+@patch("cohere.ClientV2")
 def test_cohere_embedding_operator(cohere_client, get_connection):
     """
     Test Cohere client is getting called with the correct key and that
@@ -35,12 +35,12 @@ def test_cohere_embedding_operator(cohere_client, get_connection):
         embeddings = embedded_obj
 
     api_key = "test"
-    api_url = "http://some_host.com"
+    base_url = "http://some_host.com"
     timeout = 150
     max_retries = 5
     texts = ["On Kernel-Target Alignment. We describe a family of global optimization procedures"]
 
-    get_connection.return_value = Connection(conn_type="cohere", password=api_key, host=api_url)
+    get_connection.return_value = Connection(conn_type="cohere", password=api_key, host=base_url)
     client_obj = MagicMock()
     cohere_client.return_value = client_obj
     client_obj.embed.return_value = resp
@@ -51,6 +51,6 @@ def test_cohere_embedding_operator(cohere_client, get_connection):
 
     val = op.execute(context={})
     cohere_client.assert_called_once_with(
-        api_key=api_key, api_url=api_url, timeout=timeout, max_retries=max_retries
+        api_key=api_key, base_url=base_url, timeout=timeout, max_retries=max_retries
     )
     assert val == embedded_obj

--- a/providers/tests/cohere/operators/test_embedding.py
+++ b/providers/tests/cohere/operators/test_embedding.py
@@ -37,7 +37,6 @@ def test_cohere_embedding_operator(cohere_client, get_connection):
     api_key = "test"
     base_url = "http://some_host.com"
     timeout = 150
-    max_retries = 5
     texts = ["On Kernel-Target Alignment. We describe a family of global optimization procedures"]
 
     get_connection.return_value = Connection(conn_type="cohere", password=api_key, host=base_url)
@@ -45,12 +44,8 @@ def test_cohere_embedding_operator(cohere_client, get_connection):
     cohere_client.return_value = client_obj
     client_obj.embed.return_value = resp
 
-    op = CohereEmbeddingOperator(
-        task_id="embed", conn_id="some_conn", input_text=texts, timeout=timeout, max_retries=max_retries
-    )
+    op = CohereEmbeddingOperator(task_id="embed", conn_id="some_conn", input_text=texts, timeout=timeout)
 
     val = op.execute(context={})
-    cohere_client.assert_called_once_with(
-        api_key=api_key, base_url=base_url, timeout=timeout, max_retries=max_retries
-    )
+    cohere_client.assert_called_once_with(api_key=api_key, base_url=base_url, timeout=timeout)
     assert val == embedded_obj

